app/graphql/resolvers/freaks_resolver.rb

```diff
@@ -2,11 +2,44 @@
 
 module Resolvers
   class FreaksResolver < Resolvers::Base
+    argument :filter, Types::FreakFilterType, required: false
+
     type Types::FreakType.connection_type, null: false
 
     # :reek:UtilityFunction
-    def resolve
-      Freak.all
+    def resolve(filter: nil)
+      all_of_project_ids = filter&.dig(:project_ids, :all_of)
+      any_of_project_ids = filter&.dig(:project_ids, :any_of)
+      all_of_technology_ids = filter&.dig(:technology_ids, :all_of)
+      any_of_technology_ids = filter&.dig(:technology_ids, :any_of)
+
+      query = Freak.all.group(:id).order(:id)
+
+      if all_of_project_ids.present?
+        query = query
+                .joins(:freaks_projects)
+                .having('array_agg(freaks_projects.project_id) @> ARRAY[:ids]::bigint[]', ids: all_of_project_ids)
+      end
+
+      if any_of_project_ids.present?
+        query = query
+                .joins(:freaks_projects)
+                .where(freaks_projects: { project_id: any_of_project_ids })
+      end
+
+      if all_of_technology_ids.present?
+        query = query
+                .joins(:freaks_technologies)
+                .having('array_agg(freaks_technologies.technology_id) @> ARRAY[:ids]::bigint[]', ids: all_of_technology_ids)
+      end
+
+      if any_of_technology_ids.present?
+        query = query
+                .joins(:freaks_technologies)
+                .where(freaks_technologies: { technology_id: any_of_technology_ids })
+      end
+
+      query
```

Comment:
Try to find some patterns here to re-use & extract the logic to a different class. https://refactoring.guru/smells/long-method

Reply:
Done, I've re-used code with help of another 2 classes.